crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -1490,95 +1490,96 @@ UnaryOp: ast::UnaryOp = {
-        ast::ExprAttribute { value: Box::new(e), attr, ctx: ast::ExprContext::Load, range: (location..end_location).into() }
-    ),
-};
-
-SubscriptList: ast::Expr = {
+    <location:@L> <func:AtomExpr2<"all">> <arguments:Arguments> <end_location:@R> => ast::Expr::Call(
+        ast::ExprCall { func: Box::new(func.into()), arguments, range: (location..end_location).into() }
+    ).into(),
+    <location:@L> <value:AtomExpr2<"all">> "[" <slice:SubscriptList> "]" <end_location:@R> => ast::Expr::Subscript(
+        ast::ExprSubscript { value: Box::new(value.into()), slice: Box::new(slice.into()), ctx: ast::ExprContext::Load, range: (location..end_location).into() }
+    ).into(),
+    <location:@L> <value:AtomExpr2<"all">> "." <attr:Identifier> <end_location:@R> => ast::Expr::Attribute(
+        ast::ExprAttribute { value: Box::new(value.into()), attr, ctx: ast::ExprContext::Load, range: (location..end_location).into() }
+    ).into(),
+};
+
+SubscriptList: ast::ParenthesizedExpr = {
     <location:@L> <s1:Subscript> <end_location:@R> => {
-        s1
+        s1.into()
     },
     <location:@L> <s1:Subscript> "," <end_location:@R> => {
         ast::Expr::Tuple(
             ast::ExprTuple { elts: vec![s1], ctx: ast::ExprContext::Load, range: (location..end_location).into() },
-        )
+        ).into()
     },
     <location:@L> <elts:TwoOrMore<Subscript, ",">> ","? <end_location:@R> => {
         ast::Expr::Tuple(
             ast::ExprTuple { elts, ctx: ast::ExprContext::Load, range: (location..end_location).into() },
-        )
+        ).into()
     }
 };
 
-Subscript: ast::Expr = {
+Subscript: ast::ParenthesizedExpr = {
     TestOrStarNamedExpr,
     <location:@L> <e1:Test<"all">?> ":" <e2:Test<"all">?> <e3:SliceOp?> <end_location:@R> => {
         let lower = e1.map(Box::new);
         let upper = e2.map(Box::new);
         let step = e3.flatten().map(Box::new);
         ast::Expr::Slice(
             ast::ExprSlice { lower, upper, step, range: (location..end_location).into() }
-        )
+        ).into()
     }
 };
 
 SliceOp: Option<ast::Expr> = {
     <location:@L> ":" <e:Test<"all">?> => e,
 }
 
-Atom<Goal>: ast::Expr = {
-    <location:@L> <s:(@L string @R)+> =>? Ok(parse_strings(s)?),
+Atom<Goal>: ast::ParenthesizedExpr = {
+    <location:@L> <s:(@L string @R)+> =>? Ok(parse_strings(s)?.into()),
     <location:@L> <value:Constant> <end_location:@R> => ast::Expr::Constant(
         ast::ExprConstant { value, kind: None, range: (location..end_location).into() }
-    ),
+    ).into(),
     <location:@L> <id:Identifier> <end_location:@R> => ast::Expr::Name(
         ast::ExprName { id: id.into(), ctx: ast::ExprContext::Load, range: (location..end_location).into() }
-    ),
+    ).into(),
     <location:@L> "[" <e:ListLiteralValues?> "]"<end_location:@R>  => {
-        let elts = e.unwrap_or_default();
+        let elts = e.into_iter().flatten().map(ast::Expr::from).collect();
```

Comment:
I guess this requires another allocation unfortunately, since we have to translate `Vec<ParenthesizedExpr>` to `Vec<Expr>`?